app/models/remote_theme.rb

```diff
@@ -277,6 +272,67 @@ def update_from_remote(
       end
     end
 
+    # NOTE (martin): Until we are ready to roll this out more
+    # widely, let's avoid doing this work for most sites.
+    if SiteSetting.theme_download_screenshots
```

Comment:
This whole file is in dire need of refactors...I think for now it makes sense to keep this inline like the assets collection above.